src/MainWindow/main_window.cpp

```diff
@@ -228,6 +227,39 @@ void MainWindow::cleanUpDockWidgets(std::vector<QDockWidget*>& dockWidgets) {
   dockWidgets.clear();
 }
 
+void MainWindow::openProject(const QString& project) {
+  ReShowWindow(project);
+  loadFile(project);
+}
+
+void MainWindow::saveToRecentSettings(const QString& project) {
+  if (project.isEmpty()) return;
+
+  std::list<QString> projects;
+
+  for (uint i = 0; i < 10; i++) {
```

Comment:
Consider moving the number of recent projects (10) and (5) outside this function which can be set explicitly.

Reply:
done